tests/pipeline/controlflowanalysis/test_variable_name_generation.py

```diff
@@ -74,52 +74,65 @@ def test_default_notation_1():
-    ast = AbstractSyntaxTree(CodeNode(Assignment(var := Variable("var_0", Integer(104, True)), Constant(0)), true_value), {})
+    node = CodeNode([Assignment(Variable("var_0", Integer(104, True)), Constant(0))], LogicCondition.initialize_true(LogicCondition.generate_new_context()))
+    ast = AbstractSyntaxTree(node, {})
     _run_vng(ast, _generate_options())
-    assert var._name == "unkVar0"
-
-
-def test_tmp_variable():
-    true_value = LogicCondition.initialize_true(LogicCondition.generate_new_context())
-    ast = AbstractSyntaxTree(CodeNode(Assignment(var := Variable("tmp_42", Float(64)), Constant(0)), true_value), {})
-    _run_vng(ast, _generate_options())
-    assert var._name == "dTmp42"
+    for instr in node.instructions:
+        assert instr.destination.name == "unkVar0"
 
 
 def test_same_variable():
     """Variables can be copies of the same one. The renamer should only rename a variable once. (More times would destroy the actual name)"""
-    true_value = LogicCondition.initialize_true(LogicCondition.generate_new_context())
     var1 = Variable("tmp_42", Float(64))
-    var2 = Variable("var_0", Integer(104, True))
-    ast = AbstractSyntaxTree(CodeNode([
+    node = CodeNode([
         Assignment(var1, Constant(0)),
+        Assignment(var1, Constant(0))], LogicCondition.initialize_true(LogicCondition.generate_new_context()))
+    ast = AbstractSyntaxTree(node, {})
+    _run_vng(ast, _generate_options())
+    assert node.instructions[0].destination.name == "dTmp0"
+
+def test_same_variable_idx():
+    """Variables with the same counter should not be renamed into the same thing"""
+    var1 = Variable("x_1", Integer.int32_t())
+    var2 = Variable("y_1", Integer.int32_t())
+    node = CodeNode([
         Assignment(var1, Constant(0)),
-        Assignment(var2, Constant(0)),
-        Assignment(var2, Constant(0))], true_value), {})
+        Assignment(var2, Constant(0))], LogicCondition.initialize_true(LogicCondition.generate_new_context()))
+    ast = AbstractSyntaxTree(node, {})
+    _run_vng(ast, _generate_options())
+    assert node.instructions[0].destination.name != node.instructions[1].destination.name
+
+def test_different_custom_names_0():
+    node = CodeNode([Assignment(Variable("tmp_42", Float(64)), Constant(0))], LogicCondition.initialize_true(LogicCondition.generate_new_context()))
+    ast = AbstractSyntaxTree(node, {})
+    _run_vng(ast, _generate_options())
+    assert node.instructions[0].destination.name == "dTmp0"
+
+def test_different_custom_names_1():
+    node = CodeNode([Assignment(Variable("loop_break", Float(64)), Constant(0))], LogicCondition.initialize_true(LogicCondition.generate_new_context()))
+    ast = AbstractSyntaxTree(node, {})
     _run_vng(ast, _generate_options())
-    assert var1._name == "dTmp42"
-    assert var2._name == "unkVar0"
-    
\ No newline at end of file
+    assert node.instructions[0].destination.name == "dLoopbreak0"
```

Comment:
I am not sure whether `dLoopBreak0` would be easier to read. Perhaps whenever we have a separation, the first letter should again be capitalized, but I am not sure.

Reply:
Hmm i get your point. We can talk about shortly tomorrow. Will make it like requested (but it's a bit ugly)

Reply:
I am not saying that this is better, it was a more open question. If this is the simplest and cleanest way, I am ok with it.

Reply:
My comment was a bit unclear, I like you idea better. `dLoopBreak0` is way easier to read then `dLoopbreak0`. 

My problem was rather: "How do i _cleanly_ capitalize the char after a removed one?" (with numbers/other corner cases included). Don't know if there is a better way, maybe with RE or something.